src/index.js

```diff
@@ -1,14 +1,135 @@
+  }
+  return tasks;
 }
 
-document.body.appendChild(component());
+class Task {
+  constructor(description, index) {
+    this.description = description;
+    this.completed = false;
+    this.index = index;
+  }
+
+  set updatedIndex(newIndex) {
+    this.index = newIndex;
+  }
+
+  set updatedDesc(newDesc) {
+    this.description = newDesc;
+  }
+}
+
+function prepareEdit(task, btn) {
+  btn.addEventListener('click', () => {
+    const el = document.querySelector(`[data-index="${task.index}"]`);
+    const p = el.children[1];
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.value = p.textContent;
+    input.id = task.index;
+    el.insertBefore(input, p);
+    el.removeChild(p);
+    input.focus();
+    input.select();
+    input.classList.add('edit');
+  });
+}
+
+function loadTasks() {
+  const tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
+  tasks.forEach((task) => {
+    li = document.createElement('li');
+    li.classList.add('task-li');
+    li.innerHTML = `
+        <div class="task" data-index="${task.index}">
+          <input type="checkbox">
+          <p>${task.description}</p>
+          <div class="btns">
+            <button type="button" class="close-button scroll">+</button>
+            <button type="button" class="fas fa-ellipsis-v scroll" id="edit-${task.index}"></button>
+          </div>
+        </div>
+    `;
+    ul.appendChild(li);
+    const btn = document.getElementById(`edit-${task.index}`);
+    prepareEdit(task, btn);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    getTasks();
+  });
+}
+
+function addTask() {
+  ul.innerHTML = '';
+  tasks.forEach((task) => {
+    li = document.createElement('li');
+    li.classList.add('task-li');
+    li.innerHTML = `
+        <div class="task" data-index="${task.index}">
+          <input type="checkbox">
+          <p>${task.description}</p>
+          <div class="btns">
+            <button type="button" class="close-button scroll">+</button>
+            <button type="button" class="fas fa-ellipsis-v scroll" id="edit-${task.index}"></button>
+          </div>
+        </div>
+    `;
+    ul.appendChild(li);
+    const btn = document.getElementById(`edit-${task.index}`);
+    prepareEdit(task, btn);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    getTasks();
+  });
+}
+
+function removeTask(e) {
+  if (e.target.classList.contains('close-button')) {
+    const deletedIndex = e.target.parentElement.parentElement.dataset.index;
+    const toBeUpdatedTasks = tasks.slice(deletedIndex, tasks.length);
+    toBeUpdatedTasks.forEach((task) => {
+      const el = document.querySelector(`[data-index="${task.index}"]`);
+      el.dataset.index = task.index - 1;
+      task.updatedIndex = task.index - 1;
+    });
+    e.target.parentElement.parentElement.remove();
+    tasks.splice(deletedIndex - 1, 1);
+  }
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  getTasks();
+}
```

Comment:
Please be sure to create a new js file and that you import these methods into your index.js in order to follow the project requirements 🧐